NebulaModel/Attributes/UIControlAttribute.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace NebulaModel.Attributes
+{
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public class UIControlAttribute : Attribute
+    {
+        public readonly string displayName;
+
+        public UIControlAttribute(string displayName)
+        {
+            this.displayName = displayName;
```

Comment:
Might be nitpicky, but System.ComponentModel has these attributes.

Reply:
Could use https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.displaynameattribute?view=netframework-3.5 instead.